Code/Framework/AzCore/AzCore/Memory/AllocationRecords.cpp

```diff
@@ -232,24 +238,38 @@ AllocationRecords::UnregisterAllocation(void* address, size_t byteSize, size_t a
         return;
     }
 
-    Debug::AllocationRecordsType::iterator iter = m_records.find(address);
-
-    // We cannot assert if an allocation does not exist because our allocators start up way before the driller is started and the Allocator Records would be created.
-    // It is currently impossible to actually track all allocations that happen before a certain point
-    //AZ_Assert(iter!=m_records.end(), "Could not find address 0x%p in the allocator!", address);
-    if (iter == m_records.end())
+    AllocationInfo allocationInfo;
     {
-        return;
+        AZStd::scoped_lock lock(m_recordsMutex);
+        Debug::AllocationRecordsType::iterator iter = m_records.find(address);
+        // We cannot assert if an allocation does not exist because allocations may have been made before tracking was enabled. 
+        // It is currently impossible to actually track all allocations that happen before a certain point
+        // AZ_Assert(iter!=m_records.end(), "Could not find address 0x%p in the allocator!", address);
+        if (iter == m_records.end())
+        {
+            return;
+        }
+        allocationInfo = iter->second;
+        m_records.erase(iter);
+
+        // try to be more aggressive and keep the memory footprint low.
+        // \todo store the load factor at the last rehash to avoid unnecessary rehash
+        if (m_records.load_factor() < 0.9f)
```

Comment:
So why `0.9`? for the unordered_map load factor?

Reply:
Nevermind, I see this code was moved up from below

Reply:
honestly... no idea, I imagine this came from some empiric exercise... that code was moved from the MemoryDriller